extensions/network/configurationinternal.m

```diff
@@ -328,6 +328,38 @@ static int dynamicStoreLocation(lua_State *L) {
     return 1 ;
 }
 
+/// hs.network.configuration:locations() -> table
+/// Method
+/// Returns all configured locations
+///
+/// Parameters:
+///  * None
+///
+/// Returns:
+///  * a table of key-value pairs mapping location UUIDs to their names
+///
+static int dynamicStoreLocations(lua_State *L) {
+    LuaSkin *skin = [LuaSkin shared] ;
+    [skin checkArgs:LS_TUSERDATA, USERDATA_TAG, LS_TBREAK] ;
+    SCPreferencesRef prefs = SCPreferencesCreate(NULL, CFSTR("Hammerspoon"), NULL);
```

Comment:
while it isn't *supposed* to happen, we've seen a few cases where core foundation objects which are never supposed to be NULL turn out to be NULL anyways, thus causing a crash... I would recommend wrapping from here to just before the return with `if (prefs) { ... } else { lua_pushnil(L) ; }